src/SecTester.Repeater/Bus/Formatters/MessagePackHttpHeadersFormatter.cs

```diff
@@ -0,0 +1,161 @@
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using MessagePack;
+using MessagePack.Formatters;
+
+namespace SecTester.Repeater.Bus.Formatters;
+
+// Headers formatter is to be support javascript `undefined` which is treated as null (0xC0)
+// https://www.npmjs.com/package/@msgpack/msgpack#messagepack-mapping-table
+// https://github.com/msgpack/msgpack/blob/master/spec.md#nil-format
+
+public class MessagePackHttpHeadersFormatter : IMessagePackFormatter<
```

Comment:
You can simply register formatter for `KeyValuePair<string, IEnumerable<string>>` to avoid redundant complexity and duplicating the built-in logic.

Reply:
It's need to convert map to array and vice versa.

Reply:
Not clear what you mean. Anyway,`Dictionary` also implements `IEnumerable`, just like `List`. `IEnumerable<KeyValuePair<string, IEnumerable<string>>>` is equivalent to `IDictionary<string, IEnumerable<string>>`